src/tools/web/elektrad/meta_handler.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+
+	elektra "github.com/ElektraInitiative/go-elektra/kdb"
+)
+
+type keyValueBody struct {
+	Key   string  `json:"key"`
+	Value *string `json:"value"`
+}
+
+func postMetaHandler(w http.ResponseWriter, r *http.Request) {
+	var meta keyValueBody
+
+	keyName := parseKeyNameFromURL(r)
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&meta); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	if meta.Key == "" {
+		badRequest(w)
+	}
+
+	kdb := getHandle(r)
+
+	parentKey, err := elektra.NewKey(keyName)
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	ks, err := getKeySet(kdb, parentKey)
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	k := ks.Lookup(parentKey)
+
+	if k == nil {
+		k = parentKey
+		ks.AppendKey(parentKey)
+	}
+
+	if meta.Value == nil {
+		err = k.RemoveMeta(meta.Key)
+	} else {
+		err = k.SetMeta(meta.Key, *meta.Value)
+	}
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	_, err = kdb.Set(ks, parentKey)
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	noContent(w)
+}
+
+func deleteMetaHandler(w http.ResponseWriter, r *http.Request) {
+	kdb := getHandle(r)
+
+	var meta keyValueBody
+
+	keyName := parseKeyNameFromURL(r)
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&meta); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	key, err := elektra.NewKey(keyName)
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	ks, err := getKeySet(kdb, key)
```

Comment:
This circumvents the conflict handling: `kdb.get` should only be done once in the beginning and to resolve conflicts.

Reply:
It IS only called once - or am I not seeing something?

Reply:
Maybe I do not understand something correctly. Can you describe the life-time of KDB/handlers somewhere in the docu?

Reply:
@markus2330 maybe a short call would be helpful? There's a lot going on with this PR and the go bindings and it's getting a little confusing :)

Reply:
Yes, we can phone tomorrow 10:15 in matrix?

Reply:
I would like to fix all remaining issues in both PR's today and merge them if possible

Reply:
>I would like to fix all remaining issues in both PR's today and merge them if possible


Is it done now?